osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -508,6 +508,10 @@ public override bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
                     hideOverlay(true);
                     return true;
                 }
+
+                case GlobalAction.DeselectAllMods:
+                    DeselectAll();
```

Comment:
I don't know how temporary this is supposed to be, but in an ideal world this would be triggering the "deselect all" button to flash for consistent UX, which is not immediately trivial given how everything is set up.

Reply:
How does this diff sit with you?

```diff
diff --git a/osu.Game/Overlays/Mods/ModSelectOverlay.cs b/osu.Game/Overlays/Mods/ModSelectOverlay.cs
index fb988b5bfe..1dad185b78 100644
--- a/osu.Game/Overlays/Mods/ModSelectOverlay.cs
+++ b/osu.Game/Overlays/Mods/ModSelectOverlay.cs
@@ -56,11 +56,32 @@ public abstract class ModSelectOverlay : ShearedOverlayContainer, ISamplePlaybac
         /// </summary>
         protected virtual bool ShowTotalMultiplier => true;
 
+        /// <summary>
+        /// Whether per-mod customisation controls are visible.
+        /// </summary>
+        protected virtual bool AllowCustomisation => true;
+
         protected virtual ModColumn CreateModColumn(ModType modType, Key[]? toggleKeys = null) => new ModColumn(modType, false, toggleKeys);
 
         protected virtual IReadOnlyList<Mod> ComputeNewModsFromSelection(IReadOnlyList<Mod> oldSelection, IReadOnlyList<Mod> newSelection) => newSelection;
 
-        protected virtual IEnumerable<ShearedButton> CreateFooterButtons() => createDefaultFooterButtons();
+        protected virtual IEnumerable<ShearedButton> CreateFooterButtons()
+        {
+            if (AllowCustomisation)
+            {
+                yield return customisationButton = new ShearedToggleButton(BUTTON_WIDTH)
+                {
+                    Text = ModSelectOverlayStrings.ModCustomisation,
+                    Active = { BindTarget = customisationVisible }
+                };
+            }
+
+            yield return deselectAllButton = new ShearedButton(BUTTON_WIDTH)
+            {
+                Text = CommonStrings.DeselectAll,
+                Action = DeselectAll
+            };
+        }
 
         private readonly Bindable<Dictionary<ModType, IReadOnlyList<Mod>>> availableMods = new Bindable<Dictionary<ModType, IReadOnlyList<Mod>>>();
         private readonly Dictionary<ModType, IReadOnlyList<ModState>> localAvailableMods = new Dictionary<ModType, IReadOnlyList<ModState>>();
@@ -77,6 +98,7 @@ public abstract class ModSelectOverlay : ShearedOverlayContainer, ISamplePlaybac
         private DifficultyMultiplierDisplay? multiplierDisplay;
 
         private ShearedToggleButton? customisationButton;
+        private ShearedButton? deselectAllButton;
 
         protected ModSelectOverlay(OverlayColourScheme colourScheme = OverlayColourScheme.Green)
             : base(colourScheme)
@@ -201,7 +223,7 @@ protected override void LoadComplete()
 
             // This is an optimisation to prevent refreshing the available settings controls when it can be
             // reasonably assumed that the settings panel is never to be displayed (e.g. FreeModSelectOverlay).
-            if (customisationButton != null)
+            if (AllowCustomisation)
                 ((IBindable<IReadOnlyList<Mod>>)modSettingsArea.SelectedMods).BindTo(SelectedMods);
 
             SelectedMods.BindValueChanged(val =>
@@ -256,21 +278,6 @@ private ColumnDimContainer createModColumnContent(ModType modType, Key[]? toggle
             };
         }
 
-        private ShearedButton[] createDefaultFooterButtons()
-            => new[]
-            {
-                customisationButton = new ShearedToggleButton(BUTTON_WIDTH)
-                {
-                    Text = ModSelectOverlayStrings.ModCustomisation,
-                    Active = { BindTarget = customisationVisible }
-                },
-                new ShearedButton(BUTTON_WIDTH)
-                {
-                    Text = CommonStrings.DeselectAll,
-                    Action = DeselectAll
-                }
-            };
-
         private void createLocalMods()
         {
             localAvailableMods.Clear();
@@ -510,7 +517,7 @@ public override bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
                 }
 
                 case GlobalAction.DeselectAllMods:
-                    DeselectAll();
+                    deselectAllButton?.TriggerClick();
                     return true;
             }
 
diff --git a/osu.Game/Screens/OnlinePlay/FreeModSelectOverlay.cs b/osu.Game/Screens/OnlinePlay/FreeModSelectOverlay.cs
index 6e1c9b7a59..790333ea5b 100644
--- a/osu.Game/Screens/OnlinePlay/FreeModSelectOverlay.cs
+++ b/osu.Game/Screens/OnlinePlay/FreeModSelectOverlay.cs
@@ -4,6 +4,7 @@
 using System;
 using osu.Game.Overlays;
 using System.Collections.Generic;
+using System.Linq;
 using osu.Framework.Graphics;
 using osu.Game.Graphics.UserInterface;
 using osu.Game.Overlays.Mods;
@@ -17,6 +18,8 @@ public class FreeModSelectOverlay : ModSelectOverlay
     {
         protected override bool ShowTotalMultiplier => false;
 
+        protected override bool AllowCustomisation => false;
+
         public new Func<Mod, bool> IsValidMod
         {
             get => base.IsValidMod;
@@ -31,22 +34,13 @@ public FreeModSelectOverlay()
 
         protected override ModColumn CreateModColumn(ModType modType, Key[] toggleKeys = null) => new ModColumn(modType, true, toggleKeys);
 
-        protected override IEnumerable<ShearedButton> CreateFooterButtons() => new[]
-        {
+        protected override IEnumerable<ShearedButton> CreateFooterButtons() => base.CreateFooterButtons().Prepend(
             new ShearedButton(BUTTON_WIDTH)
             {
                 Anchor = Anchor.BottomLeft,
                 Origin = Anchor.BottomLeft,
                 Text = CommonStrings.SelectAll,
                 Action = SelectAll
-            },
-            new ShearedButton(BUTTON_WIDTH)
-            {
-                Anchor = Anchor.BottomLeft,
-                Origin = Anchor.BottomLeft,
-                Text = CommonStrings.DeselectAll,
-                Action = DeselectAll
-            }
-        };
+            });
     }
 }

```

Reply:
Probably fine for the time being. I still have things in mind for those buttons which may not see this hold up in the long run, but for now it's probably okay.

Reply:
I've applied this. Also added "select all" support in the process.